<launch>

  <!-- Transformation Configuration ... Setting Up the Relationships Between Coordinate Frames -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_map" args="0 0 0 0 0 0 world map 50"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0.05 0 0 0 base_footprint base_link 50" />

  <!-- Launch RViz -->
  <include file="$(find urdf_tutorial)/launch/display.launch">
     <arg name="model" value="$(find mobile_manipulator_body)/urdf/robot_base.urdf"/>
     <arg name="rvizconfig" value="$(find mobile_manipulator)/param/mobile_manipulator.rviz"/>
  </include>
  
  <!-- Extended Kalman Filter from robot_pose_ekf Node-->
  <!-- Subscribe: /odom, /imu_data, /vo -->
  <!-- Publish: /robot_pose_ekf/odom_combined -->
  <remap from="/odom" to="/base_truth_odom" />
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom"/>
    <param name="base_footprint_frame" value="base_footprint"/>
    <param name="freq" value="50.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="gps_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
  </node>

  <!-- Map File -->
  <arg name="map_file" default="$(find mobile_manipulator)/maps/test.yaml"/>
	
  <!-- Map Server -->
  <!-- Publish: /map, /map_metadata -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" />
	
  <!-- Add AMCL example for differential drive robots for Localization -->
  <!-- Subscribe: /scan, /tf, /initialpose, /map -->
  <!-- Publish: /amcl_pose, /particlecloud, /tf -->
  <include file="$(find amcl)/examples/amcl_diff.launch"/>  
	
  <!-- Move Base Node -->
  <!-- Subscribe: /tf, /map, /scan, /odom, /move_base_simple/goal -->
  <!-- Publish: /cmd_vel -->
  <remap from="cmd_vel" to="/robot_base_velocity_controller/cmd_vel"/>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find mobile_manipulator)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mobile_manipulator)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mobile_manipulator)/param/local_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find mobile_manipulator)/param/global_costmap_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find mobile_manipulator)/param/base_local_planner_params.yaml" command="load" />
  </node>

</launch>
